data/ci/jenkins/config/config.xml

```diff
@@ -0,0 +1,283 @@
+    <githubWebUri>https://github.com</githubWebUri>
+    <githubApiUri>https://api.github.com</githubApiUri>
+    <clientID>dde1b1de183181e018b7</clientID>
+    <clientSecret>{AQAAABAAAAAwWLmVHCPZQnaZy6txJ2YqPk0kZup4D6uGbL1RDhzMSeUsTsFpkOzKa7rKY+jgUF5sVTGn/aQ8Tpnu1J3kapjgcw==}</clientSecret>
+    <oauthScopes>read:org,user:email</oauthScopes>
+  </securityRealm>
+  <disableRememberMe>false</disableRememberMe>
+  <projectNamingStrategy class="jenkins.model.ProjectNamingStrategy$DefaultProjectNamingStrategy"/>
+  <workspaceDir>${JENKINS_HOME}/workspace/${ITEM_FULLNAME}</workspaceDir>
+  <buildsDir>${ITEM_ROOTDIR}/builds</buildsDir>
+  <markupFormatter class="hudson.markup.EscapedMarkupFormatter"/>
+  <jdks/>
+  <viewsTabBar class="hudson.views.DefaultViewsTabBar"/>
+  <myViewsTabBar class="hudson.views.DefaultMyViewsTabBar"/>
+  <clouds>
+    <com.microsoft.azure.vmagent.AzureVMCloud plugin="azure-vm-agents@0.4.7.1">
+      <name>jenkins-azure-cloud</name>
+      <cloudName>jenkins-azure-cloud</cloudName>
+      <credentialsId>1073bb96-1dc9-493d-a716-5d32119156d9</credentialsId>
+      <maxVirtualMachinesLimit>10</maxVirtualMachinesLimit>
+      <resourceGroupReferenceType>existing</resourceGroupReferenceType>
+      <newResourceGroupName>jenkins-resource-group</newResourceGroupName>
+      <existingResourceGroupName>cc-jenkins-resources</existingResourceGroupName>
+      <vmTemplates>
+        <com.microsoft.azure.vmagent.AzureVMAgentTemplate>
+          <templateName>jenkins-azure-vm-ubuntu-16-04-template</templateName>
+          <templateDesc>Ubuntu 16.04 Template for Jenkins VM agents on Azure</templateDesc>
+          <labels>ubuntu_16_04_azure_cloud</labels>
+          <location>West US 2</location>
+          <virtualMachineSize>Standard_D2s_v3</virtualMachineSize>
+          <storageAccountNameReferenceType>new</storageAccountNameReferenceType>
+          <diskType>managed</diskType>
+          <newStorageAccountName>ccjenkins</newStorageAccountName>
+          <existingStorageAccountName></existingStorageAccountName>
+          <storageAccountType>Standard_LRS</storageAccountType>
+          <noOfParallelJobs>1</noOfParallelJobs>
+          <usageMode>EXCLUSIVE</usageMode>
+          <shutdownOnIdle>false</shutdownOnIdle>
+          <imageTopLevelType>advanced</imageTopLevelType>
+          <imageReferenceType>reference</imageReferenceType>
+          <builtInImage>Ubuntu 16.04 LTS</builtInImage>
+          <isInstallGit>true</isInstallGit>
+          <isInstallMaven>false</isInstallMaven>
+          <isInstallDocker>true</isInstallDocker>
+          <osType>Linux</osType>
+          <imagePublisher>Canonical</imagePublisher>
+          <imageOffer>UbuntuServer</imageOffer>
+          <imageSku>16.04-LTS</imageSku>
+          <imageVersion>latest</imageVersion>
+          <agentLaunchMethod>SSH</agentLaunchMethod>
+          <preInstallSsh>true</preInstallSsh>
+          <initScript># Install java
```

Comment:
Looking over this changeset, there are lots of what appear to be duplicated shell script snippets, which is a potential maintenance problem. Could we instead create a set of stand-alone scripts and have each `<initScript/>` section just call the appropriate script?

Reply:
I concur, it would be nice to have the scripty bits stored in a repo somewhere and not tied directly into the Jenkins UI/xml (which is how I think it currently is). I know we start with a 'bare OS image', so we will always have to have some script in Jenkins to kick off the install process, but maybe that can be more of the form:
```
apt-get install git
git pull clearcontainers/tests
./clearcontainers/tests/data/ci/jenkins/install.sh
```
or similar.
Anyway, that is probably something we can improve in the future - for now having these configs in the repo is a good first step :-)

Reply:
and, yes, having scrolled through the files, they are pretty much replications across the repos and distros - so I agree there seems to be a lot of scope for refactoring here in the future.